test/benchmarks_suite.go

```diff
@@ -38,7 +39,15 @@ func BenchmarkPeerstore(b *testing.B, factory PeerstoreFactory, variant string)
 		go addressProducer(ctx, b, p.ch, p.n)
 	}
 
-	for name, bench := range peerstoreBenchmarks {
+	// So tests are always run in the same order.
+	ordernames := make([]string, 0, len(peerstoreBenchmarks))
```

Comment:
Makes sense, although we can also change the type of `peerstoreBenchmarks` to be an array of `struct { name string, testFunc func(...) }` so the order is deterministic (I think I had it like this early on!). 

Then could do away with the sorting here and in the keybook benchmarks, and apply non-lexicographical orders manually. 

Not worth changing, but just wanted to point out an alternative.